students/alex_skrn/Lesson05Extra/logging_threading.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+"""Logging from multiple threads.
+
+To simplify things a bit I removed syslog logging and I still
+have one function only, so that not to confuse me even more.
+
+Am I moving in the right direction with what I have now?
+
+Issue 1: At this point I have troubles wrapping logging-related code into
+a function and running it repeatedly in the main block because
+in such case loggers multiply quickly and produce many repeated
+log records. This is why I keep all log definions global.
+
+Issue 2. Another unclear thing is that part about "Make sure the threads exit
+after its done with execution". Why would
+I need them after execution? I thought the purpose of threads was just
+to run some code within them. So that thread.start() launches a thread
+together with some function/program associated
+with it (i.e. threading(target=my_fun)), and as soon as the associated
+program is over, I don't need this thread any longer, isn't it?
```

Comment:
Yes, I was referring to the thread closing after it finished ITS execution, not the whole program (i.e. leaving all the threads open) You can use `import threading` and then `threading.activeCount()` to make sure you don't leave anything hanging open at the end.